search/serializers.py

```diff
@@ -333,39 +325,24 @@ class ESCourseSerializer(ESModelSerializer):
 
     object_type = COURSE_TYPE
 
-    prices = serializers.SerializerMethodField()
-    instructors = serializers.SerializerMethodField()
     topics = serializers.SerializerMethodField()
-    availability = serializers.SerializerMethodField()
-
-    course_runs = ESCourseRunSerializer(many=True, allow_null=True)
-
-    def get_prices(self, course):
-        """
-        Get the prices for a course
-        """
-        return list(course.prices.values("price", "mode"))
+    course_runs = serializers.SerializerMethodField()
 
-    def get_instructors(self, course):
+    def get_course_runs(self, instance):
         """
-        Get a list of instructor names for the course
+        Get the course runs in reverse chronological order
         """
-        return [" ".join([i.first_name, i.last_name]) for i in course.instructors.all()]
+        course_runs = instance.course_runs.all().order_by(
```

Comment:
Defining the `order_by` here is negating the `pretch_related` I think because it's creating a copy of the query for each `Course` instance. So I think what we probably want here to simplify things is to define the original query like this:


```python
from django.db.models import OuterRef, Subquery, Prefetch

Course.objects.prefetch_related(
    Prefetch(
        "course_runs",
        queryset=Subquery(
            CourseRun.objects.filter(
                course_id=OuterRef('pk')
            ).order_by(
                 "-enrollment_start", "-start_date", "-year"
            )
        )
    )
)
```

Then you can change this back to a vanilla `ESCourseRunSerializer(many=True, allow_null=True)` definition.

We'll want to apply this to both the `ViewSets` and the ES indexing code.

Also, `all()` isn't needed if you're calling at least one other `ObjectManager` method.

Reply:
I'm not sure where that query definition is supposed to go? The model definition for `course_runs` is just `course_runs = GenericRelation(CourseRun)`

Reply:
It would go in these two places:

https://github.com/mitodl/open-discussions/pull/2153/files#diff-44a0e3de85d4ee6ea1e4717ef96cad83L102
https://github.com/mitodl/open-discussions/pull/2153/files#diff-a8d441b8cd2ce3c69a44ce0d1994a23cL601

Similar code would be done for bootcamps in their related locations

Reply:
I had to change the above a bit to get it working, using a subquery was returning 0 runs:
```python
for course in Course.objects.filter(id__in=ids).prefetch_related(
        "topics",
        Prefetch(
            "course_runs",
            queryset=CourseRun.objects.order_by(
                "-enrollment_start", "-start_date", "-year"
            ),
        ),
    )
```